### PR TITLE
Feat/uat

### PR DESCRIPTION
**Pull request by Anurag-Wednesday**
### Ticket Link

---

### Related Links

---

### Description

-----Added auto deploy to S3 for Uat on PR functionality
--added workflow for the same

Link to the deployed app

http://uat-tracks.s3-website.ap-south-1.amazonaws.com/feat/UAT/

### Steps to Reproduce / Test

---

---

### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are [changed or added]
- [ ] Relevant documentation is changed or added (and PR referenced)

### GIF's

---

### CODE REVIEW

File: app/containers/HomeContainer/index.js [hunk 1]
```diff
@@ -143,6 +143,9 @@ export function HomeContainer({
       <RightContent>
         <StyledT onClick={handleStoriesClick} data-testid="redirect" id="stories" />
       </RightContent>
+      <button data-testid="redirectToTracks" onClick={() => history.push('/tracks')}>
+        Go to Tracks API
```
Comment (alichherawalla) [inline app/containers/HomeContainer/index.js:147]: react-intl

File: app/containers/HomeContainer/tests/index.test.js [hunk 2]
```diff
@@ -153,6 +153,9 @@ describe('<HomeContainer /> tests', () => {
     fireEvent.click(getByTestId('redirect'));
     await timeout(500);
     expect(historySpy).toHaveBeenCalledWith('/stories');
+    fireEvent.click(getByTestId('redirectToTracks'));
```
Comment (alichherawalla) [inline app/containers/HomeContainer/tests/index.test.js:156]: data-testids are usually kebab case

File: jest.config.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
     "!**/loadable.js",
     "!app/*/*/*/Loadable.js",
     "!**/apiUtils.js",
+    "!**/history.js",
```
Comment (alichherawalla) [inline jest.config.json:14]: this isn't right. We should be adding tests for history.js

File: app/containers/ItunesProvider/SearchContainer/index.js [hunk 4]
```diff
@@ -103,6 +107,7 @@ export function SearchContainer({
     if (!isEmpty(sTerm)) {
       dispatchGetTrackList(sTerm);
     } else {
+      message.warning('Cleared Search');
```
Comment (alichherawalla) [inline app/containers/ItunesProvider/SearchContainer/index.js:110]: react-intl please

File: app/containers/ItunesProvider/SearchContainer/index.js [hunk 5]
```diff
@@ -75,6 +75,10 @@ export function SearchContainer({
   const [playingSong, setPlayingSong] = useState(null);
   const handleOnActionClick = (track, type) => {
     if (type === 'play') {
+      notification.open({
+        message: 'Now Playing',
```
Comment (alichherawalla) [inline app/containers/ItunesProvider/SearchContainer/index.js:79]: react-intl

File: app/components/TrackDetailsComponent/index.js [hunk 6]
```diff
@@ -75,22 +81,37 @@ export function TrackDetailsComponent({
...
+            <TrackImage data-testid="carousel-image" alt={trackName} src={artworkUrl100} />
+            <TrackImage data-testid="carousel-image" alt={trackName} src={artworkUrl100} />
+            <TrackImage data-testid="carousel-image" alt={trackName} src={artworkUrl100} />
+          </StyledCarousel>
+        </Popover>
       </If>
       <Details>
         <TrackTitle>
           <If condition={!isEmpty(trackName)} otherwise={<T data-testid="no-track-name" id="no_track_name" />}>
             <T type="heading" id="track_details_name" data-testid="track_title" values={{ Track: trackName }} />
```
Comment (alichherawalla) [inline app/components/TrackDetailsComponent/index.js:106]: variable names should always be camelCase. Please rename `Track` to `track`

File: app/components/TrackDetailsComponent/index.js [hunk 7]
```diff
@@ -75,22 +81,37 @@ export function TrackDetailsComponent({
...
   return (
     <TrackDetailsCard data-testid="track-details-component">
       <If condition={!isEmpty(artworkUrl100)} otherwise={<T data-testid="image_unavailable" id="image_unavailable" />}>
-        <StyledCarousel autoplay dotPosition="bottom">
-          <TrackImage data-testid="carousel-image" alt={trackName} src={artworkUrl100} />
-          <TrackImage data-testid="carousel-image" alt={trackName} src={artworkUrl100} />
-          <TrackImage data-testid="carousel-image" alt={trackName} src={artworkUrl100} />
-        </StyledCarousel>
+        <Popover
+          title="Title"
```
Comment (alichherawalla) [inline app/components/TrackDetailsComponent/index.js:91]: title-> react-intl

File: app/components/TrackDetailsComponent/index.js [hunk 8]
```diff
@@ -75,22 +81,37 @@ export function TrackDetailsComponent({
...
+          content={
+            <T type="subheading" id="track_details_name" data-testid="track_title" values={{ Track: trackName }} />
```
Comment (alichherawalla) [inline app/components/TrackDetailsComponent/index.js:93]: track here should be in camelCase too

File: app/components/TrackComponent/index.js [hunk 9]
```diff
@@ -46,41 +46,59 @@ const Title = styled(T)`
...
-    <TrackCard
-      onClick={() => history.push(`/tracks/${trackId}`)}
-      data-testid="TrackCard"
-      title={
-        <If condition={!isEmpty(trackName)} otherwise={<T type="subheading" id="no_track_name" />}>
-          <Title type="subheading" id="name" values={{ trackName }} />
+    <>
+      <TrackCard
+        onClick={() => setVisible(true)}
+        data-testid="TrackCard"
```
Comment (Anurag-Wednesday) [inline app/components/TrackComponent/index.js:64]: snakecase

**Pull request merged**